src/graph/backend/elyzor/utils.cpp

```diff
@@ -95,6 +96,23 @@ std::vector<op_kind_t> get_supported_op_kinds() {
     return ret;
 }
 
+std::string gc_version_to_string(const dnnl_graph_compiler_version &v) {
+    std::stringstream ss;
+    ss << "API version:\n\t" << gc_version_to_string(v.api_version)
+       << "\nGC version:\n\t" << gc_version_to_string(v.gc_version);
+    return ss.str();
+}
+
+std::string gc_version_to_string(
+        const dnnl_graph_compiler_version::version &v) {
+    std::stringstream ss;
+    // Cast uint8_t to int before inserting into the stringstream
+    // as otherwise they're interpreted as characters
+    ss << static_cast<int>(v.major) << "." << static_cast<int>(v.minor) << "."
+       << static_cast<int>(v.patch) << " " << v.hash;
+    return ss.str();
+}
+
```

Comment:
Currently the `gc_version_to_string()` functions are used only to build the error message.  `std::stringstream` is created 3 times to build a string. You could avoid building the strings by passing the `std::stringstream` to these functions and printing into the same stream.

Reply:
done, defined `operator<<` for versions instead

Reply:
This is even better.